src/jit/lowerxarch.cpp

```diff
@@ -2296,6 +2296,34 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
 #endif // FEATURE_SIMD
 
 #ifdef FEATURE_HW_INTRINSICS
+//----------------------------------------------------------------------------------------------
+// IsContainableHWIntrinsicOp: Return true if 'node' is a containable HWIntrinsic op.
+//
+//  Arguments:
+//     node - The hardware intrinsic node.
+//
+bool Lowering::IsContainableHWIntrinsicOp(GenTree* node)
+{
+    if (!node->OperIsHWIntrinsic())
```

Comment:
I think this earlier return can be an assertion.

Reply:
I'm not positive. Currently this is called for any op2 which is not a containable memory op. I don't know if that will always be a HWIntrinsic node.

Reply:
I think, for example, a user defined function which returns a `Vector128<T>`, would fail such an assertion.

Reply:
I see. Thanks.

Reply:
I've not tested that, however, and @CarolEidt or @mikedn might know for certain

Reply:
Seems fine to me as it is. That's how `isContainableMemoryOp` does it as well.